src/infobox.tsx

```diff
@@ -351,3 +283,151 @@ const Shares = (props: SharesProps): JSX.Element => {
+  ]);
+
+  const [formValues, setFormState] = useState({
+    endpoint: '/',
+    file_path: '/home/' + shareProps.fileInfo.path.replace('cs3drive:', ''),
+    grantee: '',
+    idp: '',
+    role: 'viewer',
+    grantee_type: 'user'
+  });
+
+  const getUsers = ({
+    state
+  }: SelectRenderer<Record<string, any>>): Array<string> => {
+    if (state.search.length <= 0) {
+      return [];
+    }
+
+    shareProps.getUsers(state.search).then(users => {
+      const parsedUsers: any = [];
+
+      let i = 1;
+      for (const user of users) {
+        parsedUsers.push({
+          id: i++,
+          name: user.display_name,
+          displayName: user.display_name,
+          idp: user.idp,
+          grantee: user.opaque_id
+        });
+      }
+      setUserList(parsedUsers);
+    });
+
+    return [];
+  };
+
+  const setFormStateFromValues = (
+    param:
+      | React.ChangeEvent<HTMLInputElement>
+      | React.ChangeEvent<HTMLSelectElement>
+  ) => {
+    const tmpFormState: any = { ...formValues };
+    tmpFormState[param.target.name] = param.target.value;
+    setFormState(tmpFormState);
+  };
+
+  const localMakeRequest = () => {
+    const [user] = [...selectedUser];
+    const _formValues = { ...formValues };
+
+    _formValues.idp = user.idp;
+    _formValues.grantee = user.grantee;
+
+    shareProps.makeRequest(_formValues);
+  };
+
+  return (
+    <div className="jp-shareform">
+      <div className="jp-shareform-line">
+        <div className="jp-shareform-title">Grantee</div>
+        <div className="jp-shareform-element">
+          <Select
+            searchable={true}
+            options={userList}
+            values={[]}
+            create={false}
+            valueField="name"
+            labelField="displayName"
+            placeholder="Select user..."
+            onChange={(userValue: any) => {
+              const user = userValue[0] as { [key: string]: string };
+              setSelectedUser([
+                {
+                  idp: user['idp'],
+                  grantee: user['grantee']
+                }
+              ]);
+            }}
+            handleKeyDownFn={getUsers}
+          />
+        </div>
+      </div>
+      <div className="jp-shareform-line">
+        <div className="jp-shareform-title">Role</div>
+        <div className="jp-shareform-title">
+          <select onChange={setFormStateFromValues} name="role">
+            <option value="viewer">Viewer</option>
+            <option value="editor">Editor</option>
+          </select>
+        </div>
+      </div>
+
+      <button onClick={localMakeRequest}>Make request</button>
+    </div>
+  );
+};
+
+/**
+ * Main container.
+ *
+ * @constructor
+ */
+const CreateShare = (props: CreateShareProps): JSX.Element => {
+  return (
+    <>
+      <ShareForm
+        fileInfo={props.fileInfo}
+        getUsers={async (query): Promise<Array<UsersRequest>> => {
+          return await requestAPI('api/cs3/user/query?query=' + query, {});
```

Comment:
`api/` vs `/api/` bellow. We use `join`, so it doesn't matter, but at least would be more consistent.